### ISSUE TITLE
Feature Request: Add consistent IP-address or hostname which points to hostmachine in all environments

### ISSUE THREAD

**Issue by onnimonni** [docker/for-mac#1679]
### Expected behavior
172.17.42.1 points to host machine in linux default configuration.

### Actual behavior
172.17.42.1 doesn't point to host machine using docker-for-mac default configuration.

### Information
Thanks for adding `192.168.65.1` and `docker.for.mac.localhost` in the latest version of docker-for-mac. Would it be possible to add a consistent IP address or a hostname which could across all systems (linux,mac,windows)?

In my case I would want to setup our `docker-compose.yml` for development of php project for our team. Php container needs to connect back to the IDE debugger running in the host machine port 9000.

In linux I can connect to 172.17.42.1:9000 from the container and this works. But with mac I need to use a different IP and this makes our configuration needlessly complex.

For example could we change the `192.168.65.1` -> `172.17.42.1` and just change the default IP-subnet `172.*.*.*` in docker VM <-> container network? This way it would be somewhat consistent in all environments 👍

**Comment by achimnol** [comment]
I strongly agree with this idea. At least a consistent hostname (with different IPs if there are technical issues that makes it difficult) wuold be welcome.

**Comment by Mahoney** [comment]
I would also be very much in favour of providing a name that will resolve to the host cross platform. However, the name needs not to end in ```.localhost``` - see #1837 for the problems that is causing.

**Comment by craicoverflow** [comment]
+ 1

**Comment by edwardmp** [comment]
Something like docker.for.mac.localhost on Linux would be very helpful.

**Comment by aleybovich** [comment]
I second that - we need a reliable way of looping back to host for any host OS.

**Comment by jcrben** [comment]
In the meantime, it seems that you can detect if it's a Mac container and fallback in the startup script at runtime - here is how I'm doing it:
```
hostname=$(getent hosts docker.for.mac.localhost | cut -d ' ' -f 1)
if [[ -z $(grep "xdebug.remote_host" /etc/php-5.6.ini) ]]; then
  if [[ -z "$hostname" ]]; then
    hostname = ip route | grep ^default | cut -d' '' -f 3
  fi
  echo "xdebug.remote_host = $hostname" >> /etc/php-5.6.ini
fi
```

Note: someone is curious about a completely platform agnostic solution over at https://unix.stackexchange.com/questions/392992/universal-way-to-check-if-docker-for-mac-localhost-dns-name-is-resolvable

**Comment by jedwards1211** [comment]
It's a pain to merely be able to connect to a server running on the host box from selenium grid nodes  on both my local dev machine and in CI.

**Comment by achimnol** [comment]
Still, isn't there any cross-platform name for the host address?
I want to *avoid* guiding people like this:

> <img width="718" alt="image" src="https://user-images.githubusercontent.com/555156/36526305-182fd1c6-17f1-11e8-8cc0-15af1c4940c2.png">

I really hope to see improvements!

**Comment by TheDauthi** [comment]
I've been using [this script](https://git.io/vAg4S), which is based off of my answer to the stackexchange answer above. It detects the type, putting the address into `/etc/hosts` as `$DEFAULT_DOCKER_HOSTNAME`, which can be passed into the container as an environment variable (defaults to `dockerhost.internal`). It will also add any additional names you want `/etc/hosts` to have for host address that are passed in as `$EXTRA_DOCKER_HOSTNAMES`, also via environment variable. Finally, it exports `DOCKERHOST_IP` and `DOCKERHOST_OS`, which are the detected host IP and the detected host OS.

It should work in posix `sh`, busybox `ash`, `bash`, etc, unless I broke something this morning while cleaning it up before dropping it onto the gist for public consumption. `bash` has had more testing than anything else.

I do assume the environment has some variant of `ping`. Busybox ships with it. You can also use `host`, `nc`, `syslog`, `getent`, `perl`, or anything else that will call `getnameinfo` or `gethostbyname`. Even `bash` itself (4+) can do it.

I'd like to see this fixed, this is just a workaround.

**Comment by tunesmith** [comment]
I'm running into a use case where I really need to specify the host's IP address, not just the hostname alias. 

In this particular case, it's because the docker container needs an /etc/hosts entry to map a particular domain name to the host's IP address. (In this case, because it's an apache config that requires that particular domain name.) And I can't put docker.for.mac.localhost in the /etc/hosts file.

I currently have it working by adding the following to my docker-compose.yml

```
    extra_hosts:
    - "specific.domain.name:192.168.65.2"
```

But as it stands now, I gather that ip address is not predictable, and the following clearly won't work:

```
    extra_hosts:
    - "specific.domain.name:docker.for.mac.localhost"
```

Because /etc/hosts requires an ip address.

Another solution would be a mechanism to supply domain aliases of docker.for.mac.localhost (or docker.host.internal).
